packages/flutter_tools/lib/src/build_system/targets/linux.dart

```diff
@@ -122,18 +122,191 @@ class DebugBundleLinuxAssets extends Target {
     if (environment.defines[kBuildMode] == null) {
       throw MissingDefineException(kBuildMode, 'debug_bundle_linux_assets');
     }
-    final BuildMode buildMode = getBuildModeForName(environment.defines[kBuildMode]);
     final Directory outputDirectory = environment.outputDir
       .childDirectory('flutter_assets');
     if (!outputDirectory.existsSync()) {
       outputDirectory.createSync();
     }
 
-    // Only copy the kernel blob in debug mode.
-    if (buildMode == BuildMode.debug) {
-      environment.buildDir.childFile('app.dill')
-        .copySync(outputDirectory.childFile('kernel_blob.bin').path);
+    environment.buildDir.childFile('app.dill')
+      .copySync(outputDirectory.childFile('kernel_blob.bin').path);
+
+    final Depfile depfile = await copyAssets(environment, outputDirectory);
+    final DepfileService depfileService = DepfileService(
+      fileSystem: environment.fileSystem,
+      logger: environment.logger,
+    );
+    depfileService.writeToFile(
+      depfile,
+      environment.buildDir.childFile('flutter_assets.d'),
+    );
+  }
+}
+
+/// Generate an ELF binary from a dart kernel file in profile mode.
+class LinuxAotElfProfile extends AotElfBase {
+  const LinuxAotElfProfile();
+
+  @override
+  String get name => 'linux_aot_elf_profile';
+
+  @override
+  List<Source> get inputs => const <Source>[
+    Source.pattern('{FLUTTER_ROOT}/packages/flutter_tools/lib/src/build_system/targets/linux.dart'),
+    Source.pattern('{BUILD_DIR}/app.dill'),
+    Source.pattern('{PROJECT_DIR}/.packages'),
+    Source.artifact(Artifact.engineDartBinary),
```

Comment:
Likewise, `dart` isn't used by this target

Reply:
Thanks! Will fix these now.